feat: 댓글 생성 API 구현

dain391 opened the pull request:
## 이슈 번호
#5

## 작업 내용
- Comment 엔티티 정의
- CommentRequestDto, CommentResponseDto 생성
- CommentService에 댓글 생성 로직 구현
- CommentController에 /api/comments POST API 추가

## 스크린샷(선택)

dain391 committed:
feat: 댓글 생성 API 구현

- 댓글 생성 요청/응답 DTO 작성
- 댓글 엔티티 및 저장 기능 구현
- 댓글 생성용 POST API 컨트롤러 구현

resolves: #5

dain391 committed:
refactor: 컨트롤러 코드 수정 및 ApiResponse 적용

resolves: #5

Jaehyeon-Han commented on src/main/java/com/sparta/taskflow/domain/comment/entity/Comment.java:
```diff
@@ -0,0 +1,37 @@
+
+    private String content;
+
+    private Long taskId;
```
연관관계로 안 해도 될지 논의해봐야할 것 같아요

Jaehyeon-Han commented on src/main/java/com/sparta/taskflow/domain/comment/entity/Comment.java:
```diff
@@ -0,0 +1,37 @@
+@Entity
+@Getter
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+public class Comment {
```
BaseTimeEntity 상속해주세요. DB 스키마가 잘못 올라와있는 것 같아 확인 후 반영해야 할 것 같아요.

Jaehyeon-Han commented on src/main/java/com/sparta/taskflow/domain/comment/dto/CommentRequestDto.java:
```diff
@@ -0,0 +1,13 @@
+
+@Getter
+@NoArgsConstructor
+public class CommentRequestDto {
```
API 상에는 Long userId 대신 String author가 들어오는 것으로 돼있습니다. 수정 부탁드려요.

GyeongSe99 replied:
바꾼다면 명세서를 Long userId를 받도록 수정하는게 좋을 것 같아요

Jaehyeon-Han commented on src/main/java/com/sparta/taskflow/domain/comment/dto/CommentRequestDto.java:
```diff
@@ -0,0 +1,13 @@
+
+@Getter
+@NoArgsConstructor
+public class CommentRequestDto {
```
이름도 CreateCommentRequestDto처럼 어떤 작업에 대한 요청 DTO인지 나타내주는 것이 좋을 것 같아요.

GyeongSe99 commented on src/main/java/com/sparta/taskflow/domain/comment/entity/Comment.java:
```diff
@@ -0,0 +1,37 @@
+
+@Entity
+@Getter
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
```
혹시 생성자의 접근을 protected로 제한한 이유가 무엇인가요?

dain391 replied:
@NoArgsConstructor만 사용하면 기본 생성자의 접근 제어자가 public이 되기 때문에, 실수로 필드가 비어 있는 객체를 외부에서 생성할 위험이 있습니다.
그래서 access = AccessLevel.PROTECTED를 설정해서, JPA는 내부적으로 사용할 수 있도록 하고, 외부에서는 불완전한 객체 생성을 방지하려고 제한을 두었습니다.

dain391 committed:
refactor: 댓글 생성 기능 피드백 반영 및 BaseTimeEntity 상속 적용

- Comment 엔티티에 BaseTimeEntity 상속
- 생성자에서 LocalDateTime 제거
- CreateCommentRequestDto로 클래스명 명확히 변경
- userId 필드는 유지하며 명세서와 일치하도록 대응
- 연관관계 관련 논의 예정

resolves: #5

dain391 committed:
chore: task 연관관계 매핑 예정 위치에 주석 추가

resolves: #5
